schemas/applications/schema/events/set.schema.json

```diff
@@ -15,40 +15,97 @@
     }
   ],
   "$defs": {
+    "parameter": {
+      "title": "Parameter Name",
+      "description": "The name of a parameter to set",
+      "type": "string",
+      "pattern": "(^[a-zA-Z][a-zA-Z0-9_]*[a-zA-Z0-9]$)|(^[a-zA-Z]$)"
+    },
+    "value": {
+      "title": "Parameter Value",
+      "description": "The value of the parameter to set",
+      "type": [
+        "boolean",
+        "number",
+        "string",
+        "array"
+      ],
+      "items": {
+        "type": [
+          "boolean",
+          "number",
+          "string"
+        ]
+      }
+    },
+    "controller": {
+      "title": "Conntroller Name", 
+      "description": "The name of a controller",
+      "type": "string",
+      "pattern": "(^[a-zA-Z][a-zA-Z0-9_]*[a-zA-Z0-9]$)|(^[a-zA-Z]$)"
+    },
     "set_parameter_object": {
-      "type": "object",
-      "additionalProperties": false,
-      "properties": {
-        "parameter": {
-          "title": "Parameter Name",
-          "description": "The name of a parameter to set",
-          "type": "string",
-          "pattern": "(^[a-zA-Z][a-zA-Z0-9_]*[a-zA-Z0-9]$)|(^[a-zA-Z]$)"
+      "oneOf": [
+        {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "parameter": {
+              "$ref": "#/$defs/parameter"
+            },
+            "value": {
+              "$ref": "#/$defs/value"
+            }
+          },
+          "required": [
+            "parameter",
+            "value"
+          ]
         },
```

Comment:
Instead of having two definitions for the component parameter (one without a component, and one with), just one should suffice. The component field is not required, so just define it but don't include it in the "required" list
```suggestion
```

Reply:
Yes, true. Thanks